lib/Frontend/CachingUtils.cpp

```diff
@@ -436,6 +437,17 @@ CachingTool::CachingTool(StringRef Path) {
   Cache = std::move(DB->second);
 }
 
+CachingTool::CachingTool(clang::CASOptions CASOpts) {
+  auto DB = CASOpts.getOrCreateDatabases();
+  if (!DB) {
+    llvm::errs() << "Failed to create CAS: " << toString(DB.takeError())
```

Comment:
Errors should propagate to the caller via an API. The build system eventually will need to report them to the user and what you write to`stderr` doesn't necessarily become visible to the user.

Reply:
Put error into the diagnostics of dependency scanner so it is even cleaner on API surface.